sdk/identity/Azure.Identity/src/SharedTokenCacheCredential.cs

```diff
@@ -126,6 +129,11 @@ private async ValueTask<AccessToken> GetTokenImplAsync(bool async, TokenRequestC
 
         private async Task<IAccount> GetAccountAsync()
         {
+            if (_record != null)
+            {
+                return (AuthenticationAccount)_record;
```

Comment:
Is this cast required?

Reply:
Yes, [AuthenticationRecord](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/src/AuthenticationRecord.cs) doesn't implement `IAccount` directly, because we didn't want to leak this MSAL type through our abstraction. The internal type [AuthenticationAccount](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/identity/Azure.Identity/src/AuthenticationAccount.cs) implements `IAccount` and there is are [explicit cast operators](https://github.com/Azure/azure-sdk-for-net/blob/30f35c163491f262fa5769b62fad98da2254bd86/sdk/identity/Azure.Identity/src/AuthenticationAccount.cs#L22) to go back and forth.